packages/components-js/tests/vrt/playwright/specs/states/pin-code.vrt.ts

```diff
@@ -0,0 +1,98 @@
+import { expect, type Page, test } from '@playwright/test';
+import {
+  baseSchemes,
+  baseThemes,
+  baseViewportWidth,
+  forceFocusHoverState,
+  forceFocusState,
+  forceHoverState,
+  getPlaygroundPseudoStatesMarkup,
+  type PrefersColorScheme,
+  setContentWithDesignSystem,
+} from '../../helpers';
+import { type Theme } from '@porsche-design-system/utilities-v2';
+import { currentInputId } from '@porsche-design-system/components/src/components/pin-code/pin-code-utils';
+
+const component = 'pin-code';
+
+const scenario = async (page: Page, theme: Theme, scheme?: PrefersColorScheme): Promise<void> => {
+  const markup = () => `
+    <p-pin-code theme="${theme}" label="input gets hovered or focused"></p-pin-code>
+    <p-pin-code class="force-label" theme="${theme}" label="label gets hovered or focused"></p-pin-code>
+    <p-pin-code theme="${theme}" label="Disabled" disabled="true"></p-pin-code>
+    <p-pin-code theme="${theme}" label="Error state" state="error" message="Some error validation message."></p-pin-code>
+    <p-pin-code theme="${theme}" label="Success state" state="success" message="Some success validation message."></p-pin-code>
+    <p-pin-code theme="${theme}">
+      <span slot="label">
+        Slotted label
+        <span>
+          and some slotted, deeply nested <a href="#">anchor</a>.
+        </span>
+      </span>
+    </p-pin-code>
+    <p-pin-code theme="${theme}" label="Some label" state="error">
+      <span slot="description">
+        Slotted description
+        <span>
+          and some slotted, deeply nested <a href="#">anchor</a>.
+        </span>
+      </span>
+    </p-pin-code>
+    <p-pin-code theme="${theme}" label="Some label" state="error">
+      <span slot="message">
+        Slotted error message
+        <span>
+          and some slotted, deeply nested <a href="#">anchor</a>.
+        </span>
+      </span>
+    </p-pin-code>
+    <p-pin-code theme="${theme}" label="Some label" state="success">
+      <span slot="message">
+        Slotted success message
+        <span>
+          and some slotted, deeply nested <a href="#">anchor</a>.
+        </span>
+      </span>
+    </p-pin-code>`;
+
+  await setContentWithDesignSystem(page, getPlaygroundPseudoStatesMarkup(markup), {
+    forceComponentTheme: theme,
+    prefersColorScheme: scheme,
+  });
+
+  await forceHoverState(page, '.hover p-pin-code:not(.force-label) >>> input');
+  await forceHoverState(page, '.hover p-pin-code.force-label >>> span');
+  await forceHoverState(page, `.hover p-pin-code.force-label >>> #${currentInputId}`);
```

Comment:
Would be good to have a generic and reusable solution for this where the value of the `for` attribute is read and then the target is selected dynamically. 
Or is this part of the follow up ticket of other components?

Reply:
I found a more generic solution, but i thin a generic util should be created within the follow-up ticket since pin code is currently the only relevant component